src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -58,16 +74,14 @@ protected Void doInBackground() throws Exception {
 			if (this.stopGenerating) {
 				break;
 			}
-
-			HttpMessage msg = this.httpMessage.cloneRequest();
-
+			HttpMessage msgClone = msg.cloneRequest();
+			Thread.sleep(51);

```

Comment:
We should create another option to not create a dependency between the two add-ons.